xla/mlir/backends/gpu/transforms/lmhlo_gpu_to_gpu_runtime.cc

```diff
@@ -747,63 +750,92 @@ class FusedAttentionBackwardLowering
   LogicalResult matchAndRewrite(FusedDotAttentionBackward op,
                                 PatternRewriter& rewriter) const override {
     // Get the custom call target.
-    std::string fused_attention = kCustomCallTarget;
+    bool is_flash_attention = op.getIsFlashAttention();
+    std::string fused_attention = is_flash_attention
+                                      ? kFlashAttentionCustomCallTarget
+                                      : kFusedAttentionCustomCallTarget;
     auto num_operands = op.getNumOperands();
-    switch (op.getFusedMhaDag()) {
-      case mlir::lmhlo_gpu::FusedMhaBackwardDagSignature::
-          BackwardScaleBiasSoftmax:
-        if (num_operands == 10) {
-          fused_attention += "scale.softmax";
-        } else if (num_operands == 11) {
-          fused_attention += "scale.dbias.softmax";
-        } else {
-          return op.emitOpError(
-              "unexpected number of operands for fused attention backward - "
-              "BMM_Bias_Softmax_BMM");
-        }
-        break;
-
-      case mlir::lmhlo_gpu::FusedMhaBackwardDagSignature::
-          BackwardScaleBiasSoftmaxDropout:
-        if (num_operands == 10) {
-          fused_attention += "scale.softmax.dropout";
-        } else if (num_operands == 11) {
-          fused_attention += "scale.dbias.softmax.dropout";
-        } else {
-          return op.emitOpError(
-              "unexpected number of operands for fused attention backward - "
-              "BMM_Bias_Softmax_Dropout_BMM");
-        }
-        break;
-
-      case mlir::lmhlo_gpu::FusedMhaBackwardDagSignature::
-          BackwardScaleBiasMaskSoftmax:
-        if (num_operands == 11) {
-          fused_attention += "scale.mask.softmax";
-        } else if (num_operands == 12) {
-          fused_attention += "scale.dbias.mask.softmax";
-        } else {
-          return op.emitOpError(
-              "unexpected number of operands for fused attention backward - "
-              "BMM_Bias_Mask_Softmax_BMM");
-        }
-        break;
-
-      case mlir::lmhlo_gpu::FusedMhaBackwardDagSignature::
-          BackwardScaleBiasMaskSoftmaxDropout:
-        if (num_operands == 11) {
-          fused_attention += "scale.mask.softmax.dropout";
-        } else if (num_operands == 12) {
-          fused_attention += "scale.dbias.mask.softmax.dropout";
-        } else {
-          return op.emitOpError(
-              "unexpected number of operands for fused attention backward - "
-              "BMM_Bias_Mask_Softmax_Dropout_BMM");
-        }
-        break;
-
-      default:
-        return op.emitOpError("Undefined fused attention DAG signature");
+    if (!is_flash_attention) {
```

Comment:
Maybe it's possible to minimize nesting/diff by returning early?

Reply:
fixed